--- a/arch/arm/boot/dts/sun8i-h3-nanopi-neo.dts	2018-04-18 20:54:12.000000000 -1000
+++ b/arch/arm/boot/dts/sun8i-h3-nanopi-neo.dts	2018-04-22 21:19:51.510963850 -1000
@@ -47,6 +47,14 @@
 	compatible = "friendlyarm,nanopi-neo", "allwinner,sun8i-h3";
 };

+&codec {
+	allwinner,audio-routing =
+		"Line Out", "LINEOUT",
+		"MIC1", "Mic",
+		"Mic",  "MBIAS";
+	status = "okay";
+};
+
 &ehci0 {
 	status = "okay";
 };
