--- a/arch/arm64/boot/dts/allwinner/sun50i-h5-orangepi-pc2.dts	2020-10-16 20:31:22.000000000 -1000
+++ b/arch/arm64/boot/dts/allwinner/sun50i-h5-orangepi-pc2.dts	2020-10-18 07:31:59.454720485 -1000
@@ -98,6 +98,18 @@
 	cpu-supply = <&reg_vdd_cpux>;
 };

+&cpu1 {
+	cpu-supply = <&reg_vdd_cpux>;
+};
+
+&cpu2 {
+	cpu-supply = <&reg_vdd_cpux>;
+};
+
+&cpu3 {
+	cpu-supply = <&reg_vdd_cpux>;
+};
+
 &de {
 	status = "okay";
 };
@@ -179,7 +191,7 @@
 	reg_vdd_cpux: regulator@65 {
 		compatible = "silergy,sy8106a";
 		reg = <0x65>;
-		regulator-name = "vdd-cpux";
+		regulator-name = "reg_vdd_cpux";
 		silergy,fixed-microvolt = <1100000>;
 		regulator-min-microvolt = <1000000>;
 		regulator-max-microvolt = <1400000>;
