--- a/arch/arm64/boot/dts/allwinner/sun50i-h5-nanopi-neo-plus2.dts	2020-10-16 20:31:22.000000000 -1000
+++ b/arch/arm64/boot/dts/allwinner/sun50i-h5-nanopi-neo-plus2.dts	2020-10-18 07:29:01.380798791 -1000
@@ -54,9 +54,9 @@
 		regulator-max-microvolt = <3300000>;
 	};

-	vdd_cpux: gpio-regulator {
+	reg_vdd_cpux: gpio-regulator {
 		compatible = "regulator-gpio";
-		regulator-name = "vdd-cpux";
+		regulator-name = "reg_vdd_cpux";
 		regulator-type = "voltage";
 		regulator-boot-on;
 		regulator-always-on;
@@ -75,6 +75,22 @@
 	};
 };

+&cpu0 {
+	cpu-supply = <&reg_vdd_cpux>;
+};
+
+&cpu1 {
+	cpu-supply = <&reg_vdd_cpux>;
+};
+
+&cpu2 {
+	cpu-supply = <&reg_vdd_cpux>;
+};
+
+&cpu3 {
+	cpu-supply = <&reg_vdd_cpux>;
+};
+
 &codec {
 	allwinner,audio-routing =
 		"Line Out", "LINEOUT",
